parser/repository.go

```diff
@@ -0,0 +1,52 @@
+package parser
+
+import (
+	"sync"
+
+	"github.com/activecm/rita/pkg/beaconproxy"
+	"github.com/activecm/rita/pkg/certificate"
+	"github.com/activecm/rita/pkg/host"
+	"github.com/activecm/rita/pkg/hostname"
+	"github.com/activecm/rita/pkg/uconn"
+	"github.com/activecm/rita/pkg/useragent"
+)
+
+// ParseResults contains the data which the analysis packages
+// expect from the parser as well as locks for safely
+// accessing the data from multipel goroutines.
+type ParseResults struct {
```

Comment:
`ParseResults` is a struct which holds all of the result maps from the previous version of the parser. Additionally, each map is associated with an individual mutex in order to provide more granular thread safety.

https://github.com/activecm/rita/commit/05627e59d3c2f7e27fb48cffa74534c8eab839d3